diff -r 6721e038257a -r 15cc1c8a6371 hotspot/src/cpu/aarch64/vm/interp_masm_aarch64.hpp
--- a/hotspot/src/cpu/aarch64/vm/interp_masm_aarch64.hpp	Thu Dec 13 01:15:21 2018 +0000
+++ b/hotspot/src/cpu/aarch64/vm/interp_masm_aarch64.hpp	Tue Feb 04 04:42:57 2020 +0000
@@ -38,8 +38,6 @@
  protected:
 
  protected:
-  using MacroAssembler::call_VM_leaf_base;
-
   // Interpreter specific version of call_VM_base
   using MacroAssembler::call_VM_leaf_base;
 
