--- jdk8-jdk8-b132.orig/hotspot/make/linux/makefiles/adjust-mflags.sh	2015-09-17 14:17:39.000000000 +0200
+++ jdk8-jdk8-b132/hotspot/make/linux/makefiles/adjust-mflags.sh	2015-09-17 14:19:01.000000000 +0200
@@ -64,7 +64,7 @@
 	echo "$MFLAGS" \
 	| sed '
 		s/^-/ -/
-		s/ -\([^ 	][^ 	]*\)j/ -\1 -j/
+		s/ -\([^ 	][^ 	]*\)j\([0-9][0-9]*\)/ -\1 -j\2/
 		s/ -j[0-9][0-9]*/ -j/
 		s/ -j\([^ 	]\)/ -j -\1/
 		s/ -j/ -j'${HOTSPOT_BUILD_JOBS:-${default_build_jobs}}'/
