--- jdk8-jdk8-b132/jdk/src/share/native/com/sun/java/util/jar/pack/utils.cpp.orig	2015-09-17 14:58:44.000000000 +0200
+++ jdk8-jdk8-b132/jdk/src/share/native/com/sun/java/util/jar/pack/utils.cpp	2015-09-17 14:59:36.000000000 +0200
@@ -81,7 +81,7 @@
 int assert_failed(const char* p) {
   char message[1<<12];
   sprintf(message, "@assert failed: %s\n", p);
-  fprintf(stdout, 1+message);
+  fputs(1+message, stdout);
   breakpoint();
   unpack_abort(message);
   return 0;
--- jdk8-jdk8-b132/jdk/src/share/native/com/sun/java/util/jar/pack/unpack.cpp.orig	2015-09-17 14:57:57.000000000 +0200
+++ jdk8-jdk8-b132/jdk/src/share/native/com/sun/java/util/jar/pack/unpack.cpp	2015-09-17 15:01:09.000000000 +0200
@@ -3712,7 +3712,7 @@
   default:
     if (nrefs == 0) {
       buf = getbuf(20);
-      sprintf((char*)buf.ptr, TAG_NAME[tag]);
+      strcpy((char*)buf.ptr, TAG_NAME[tag]);
     } else if (nrefs == 1) {
       return refs[0]->string();
     } else {
