--- jdk9u-jdk-9.0.4+12/hotspot/src/os/linux/vm/os_linux.cpp.orig	2020-08-16 13:46:08.240275890 +0200
+++ jdk9u-jdk-9.0.4+12/hotspot/src/os/linux/vm/os_linux.cpp	2020-08-16 13:47:54.932632691 +0200
@@ -289,9 +289,9 @@
   //        ...
   //        7: The default directories, normally /lib and /usr/lib.
 #if defined(AMD64) || (defined(_LP64) && defined(SPARC)) || defined(PPC64) || defined(S390)
-  #define DEFAULT_LIBPATH "/usr/lib64:/lib64:/lib:/usr/lib"
+  #define DEFAULT_LIBPATH "/usr/lib64/java:/usr/lib64:/lib64:/lib:/usr/lib"
 #else
-  #define DEFAULT_LIBPATH "/lib:/usr/lib"
+  #define DEFAULT_LIBPATH "/usr/lib/java:/lib:/usr/lib"
 #endif
 
 // Base path of extensions installed on the system.
