--- jdk8u-jdk8u92-b31/jdk/src/share/native/sun/misc/URLClassPath.c.orig	2016-07-29 15:33:17.499988085 +0200
+++ jdk8u-jdk8u92-b31/jdk/src/share/native/sun/misc/URLClassPath.c	2016-07-29 15:35:59.085075633 +0200
@@ -24,6 +24,7 @@
  */
 
 #include <string.h>
+#include <stdlib.h>
 
 #include "jni.h"
 #include "jni_util.h"
@@ -36,6 +37,12 @@
 extern char*
 getUTF(JNIEnv *env, jstring str, char* localBuf, int bufSize);
 
+extern JNIEXPORT jboolean
+VerifyClassname(char *name, jboolean allowArrayClass);
+
+extern JNIEXPORT jboolean
+VerifyFixClassname(char *name);
+
 
 JNIEXPORT jboolean JNICALL
 Java_sun_misc_URLClassPath_knownToNotExist0(JNIEnv *env, jclass cls, jobject loader,

